Add first draft of Git hooks (generic Git) usage

ddbeck opened the pull request.

ddbeck commented on pages/integrations/git.md.erb:
```diff
@@ -0,0 +1,92 @@
+
+You can trigger builds using Git hook scripts and the Buildkite REST API.
+
+This page explains how to trigger builds when you push to a Git server.
```
> page

I suppose this could also be a "topic" or "guide" or something else. I'm not sure what's conventional around here.

ddbeck commented on pages/integrations/git.md.erb:
```diff
@@ -0,0 +1,92 @@
+
+{:toc}
+
+## Before you start
```
> Before you start

OK, this is strictly speaking unconventional — "prerequisites" seems to be the done thing around here — but it doesn't have to be. Just saying.

plaindocs replied:
I count 3 `prerequisites` and 1 `before you start`, we can easily make this change.
I think often we just put it in body text, but I'm +1 on being explicit.

ddbeck replied:
Replaced the other 3 with ee886671. Can cherry-pick to a separate PR if this unwanted here.

ddbeck reviewed (commented):
Self-review to guide other reviewers.

plaindocs commented on pages/integrations/git.md.erb:
```diff
@@ -0,0 +1,94 @@
+
+To use this as written, be sure to:
+
+- Set a valid `BUILDKITE_API_TOKEN`, `BUILDKITE_ORG_SLUG`, and `BUILDKITE_PIPELINE_SLUG`, or replace them with environment variables.
```
Link this to https://buildkite.com/docs/apis/managing-api-tokens maybe?

ddbeck commented on pages/integrations/git.md.erb:
```diff
@@ -0,0 +1,94 @@
+You can trigger builds using Git hook scripts and the Buildkite REST API.
+
+This page explains how to trigger builds when you push to a Git server.
+For example, if you're using a proprietary Git server, then you can trigger builds on push with a post-recieve hook.
```
What do you think of "If your Git server isn't an integrated repository provider" in 2e9529e7? I'm not sure we need to give the exploded view of _why_ other servers might not be directly supported, just that they're not?

ddbeck commented on pages/integrations/git.md.erb:
````diff
@@ -0,0 +1,94 @@
+```bash
+#!/usr/bin/env bash
+
+BUILDKITE_API_TOKEN="secret1234token"
````
I like what you've done, though I don't think linking that page is a good idea in this case (we're not running the script from inside a pipeline, so a lot of the assumptions mentioned there aren't so helpful). That said, we _can_ explicitly call for an environment variable, so readers don't end up distributing API keys across across a bunch of scripts. cfcdfd84

plaindocs replied:
Yeah, checks out. But let's highlight that this is info you want to keep secret, so suggest a secure method, whatever it is?

ddbeck replied:
Understood. I cribbed some wording from https://buildkite.com/docs/pipelines/secrets#using-a-secrets-storage-service with 8eadb06, so I think we're good here.

ddbeck commented on pages/integrations/git.md.erb:
````diff
@@ -0,0 +1,94 @@
+done
+```
+
+To use this as written, be sure to:
````
Not to mention needlessly wordy. Fixed with 2e9529e7.

ddbeck commented on pages/integrations/git.md.erb:
```diff
@@ -0,0 +1,94 @@
+
+    If your Git server is hosted on a platform that restricts or prohibits running arbitrary scripts, such as GitHub, then this approach won't work.
+
+- Familiarity with the concepts of executable shell scripts, Buildkite pipelines and builds, and REST APIs
```
🤦‍♂️ Fixed with fde85bad.

ddbeck commented:
> Other Git servers is the least worst

Agreed. I've updated to that option in 6a54da2a.

> It feels a little odd to send folks to learn more only to external GH related pages. Could send them to Writing Build Scripts or something as well?

Yeah, it felt a little odd at the time. I think Writing Build Scripts wasn't a great fit (it makes a bunch of assumptions about running on an agent, which we aren't), but linking to the Builds API page seemed right. An industrious developer could cancel builds, add metadata, etc. a8d36756

> we have a super secret landing page config

Linked in 13583353. I also added Phabricator on the assumption that the sidebar and landing page lists should be the same. Let me know if this was an overreach.

And that's all I've got now, @plaindocs. Ready for re-review and/or commentary from Keith.

plaindocs reviewed (approved):
Nice work on the fixes. 

This git hook isn't running on a agent, but any builds we trigger using it will, so I reckon 'Build Scripts' would be valid, but I don't care enough to suggest going back and changing it. And :+1: to the API page.

The landing page doesn't have to mirror the side bar, in fact it gives us an opportunity to highlight different facets. That said, sure let's put Phabricator in there. :+1: 

Let's not block on Keith, so when you're happy with it, hit it! :shipit:
